index.ts

```diff
@@ -18,8 +18,48 @@ export class Tree<TValue extends number> {
   }
 
   private _insert(node: ITreeNode<TValue>, value: TValue) {
-    const direction = value >= node.value ? 'right' : 'left'
-    if (node[direction] !== undefined) this._insert(node[direction]!, value) // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    const direction = this._setDirection(node, value)
+    /* eslint-disable */
+    if (node[direction] !== undefined) this._insert(node[direction]!, value)
     else node[direction] = { value }
+    /* eslint-enable */
+  }
+
+  private _setDirection(node: ITreeNode<TValue>, value: TValue) {
+    return value >= node.value ? 'right' : 'left'
+  }
+
+  public remove(value: TValue) {
+    if (this.root === undefined) throw new Error('There is no tree currently')
+    /* eslint-disable */
+    if (value === this.root!.value!) {
+      /* eslint-enable */
+      this.root = undefined
+    } else {
+      this._remove(this.root, value)
+    }
+  }
+
+  private _remove(node: ITreeNode<TValue>, value: TValue) {
+    const direction = this._setDirection(node, value)
+    /* eslint-disable */
+    if (node[direction] !== undefined && node[direction]!.value !== value)
```

Comment:
```suggestion
    if (node[direction]?.value !== value)
```

Reply:
Jest: "global" coverage threshold for branches (100%) not met: 96.42%

with this modification, the test coverage on branch will not be 100%